### PR TITLE
Allow solution to be deployed into an existing VPC

### LINKED ISSUE #169: Should be able to deploy in existing vpc
provide capability to deploy in existing vpc. Also wondering if we look as control tower architecture, this utility where should we deploy?
1. Shared Network Account
2. Shared services Account
3. Security and audit Account

### PR DESCRIPTION
**Pull request by svozza**
Issue #, if available:
Resolves #169.

Description of changes:

This PR adds new parameters to the main CloudFormation template to allow users to provide the VPC ID of an existing VPC  to deploy the solution into (the user must also supply the VPC's subnets IDs and CIDR block). If the VPC ID parameter is left blank then the template provisions a VPC for the solution.

### CODE REVIEW

File: source/cfn/templates/main.template [hunk 1]
```diff
@@ -173,11 +173,95 @@ Parameters:
...
+    RuleCondition: !Not [!Equals [!Ref ConfigAggregatorName, '']]
+    Assertions:
+      - Assert: !Equals [!Ref CrossAccountDiscovery, AWS_ORGANIZATIONS]
+        AssertDescription: | 
+          A config aggregator can only be specified when cross account discovery is managed by AWS
+          Organizations
+  VpcRule:
+    RuleCondition: !Not [!Equals [!Ref VpcId, '']]
+    Assertions:
+      - Assert: !Not [!And [!Equals [ !Ref PrivateSubnet0, '' ], !Equals [ !Ref PrivateSubnet1, '' ]]]
```
Comment (matteofigus) [inline source/cfn/templates/main.template:260]:
```suggestion
      - Assert: !Not [!Or [!Equals [ !Ref PrivateSubnet0, '' ], !Equals [ !Ref PrivateSubnet1, '' ]]]
```
If you need both subnets not to be empty, shouldn't this `!And` be an `!Or`?

File: source/cfn/templates/neptune.template [hunk 2]
```diff
@@ -282,7 +276,6 @@ Resources:
   NeptuneDBInstance:
     Type: AWS::Neptune::DBInstance
     Properties:
-      AvailabilityZone: !Ref AvailabilityZone0
```
Comment (matteofigus) [inline source/cfn/templates/neptune.template:285]: What's the consequence of omitting this? I see you are still fetching the AZ from a static mapping in your VPC, shouldn't this be more like the following to not break existing deployments?

```
AvailabilityZone: !If
  - CreateVpc
  - !Ref AvailabilityZone0
  - !Ref AWS::NoValue
```

**Review by matteofigus — COMMENTED** [review]
A couple of questions - I am not extremely familiar with the codebase, so I based these on just reviewing the CloudFormation code without testing.

File: source/cfn/templates/main.template (hunk 1 above)
[the remark this answers is not in the document]
Comment (svozza) [reply source/cfn/templates/main.template:260]: This confused me too. I initially had it as `!Or` and it didn't work as expected when I tested it.

File: source/cfn/templates/discovery-crawler.template [hunk 3]
```diff
@@ -368,7 +373,6 @@ Resources:
           Id: apiScheduledTask
           EcsParameters:
             TaskDefinitionArn: !Ref TaskDefinition
-            LaunchType: FARGATE
```
[the remark this answers is not in the document]
Comment (svozza) [reply source/cfn/templates/discovery-crawler.template:371]: You can't specify `LaunchType` and `CapacityProviderStrategy` and the same time, I got an error when I tried to deploy it and I'm not sure how it worked before.

> InvalidParameterException: Specifying both a launch type and capacity provider strategy is not supported. Remove one and try again.

File: source/cfn/templates/neptune.template (hunk 2 above)
Comment (svozza) [reply source/cfn/templates/neptune.template:285]: Ah yes, good catch!

Comment (svozza) [reply source/cfn/templates/neptune.template:285]: *fixed

File: source/cfn/templates/main.template [hunk 4]
```diff
@@ -282,12 +368,10 @@ Resources:
       TimeoutInMinutes: 60
       Parameters:
         AppName: !FindInMap [Solution, Constants, AppName]
-        AvailabilityZone0: !GetAtt VpcStack.Outputs.AvailabilityZone0
-        AvailabilityZone1: !GetAtt VpcStack.Outputs.AvailabilityZone1
```
Comment (matteofigus) [inline source/cfn/templates/main.template:286]: You did a fix in the neptune stack, but you need to fix this one too? Something like `!If [CreateVpc, ...`

**Review by matteofigus — COMMENTED** [review]
Another small issue

File: source/cfn/templates/main.template (hunk 4 above)
Comment (svozza) [reply source/cfn/templates/main.template:286]: Ah yes, good point!

Comment (svozza) [reply source/cfn/templates/main.template:286]: *done

File: .idea/aws.xml [hunk 5]
```diff
@@ -2,14 +2,15 @@
 <project version="4">
   <component name="accountSettings">
     <option name="activeProfile" value="profile:perspective-svozza" />
-    <option name="activeRegion" value="us-east-1" />
+    <option name="activeRegion" value="eu-west-1" />
     <option name="recentlyUsedProfiles">
       <list>
         <option value="profile:perspective-svozza" />
```
Comment (matteofigus) [inline .idea/aws.xml:8]: Should this file be gitignored?

File: source/cfn/templates/main.template (hunk 1 above)
[the remark this answers is not in the document]
Comment (matteofigus) [reply source/cfn/templates/main.template:260]: Ok, if you're happy then I'll resolve this.

**Review by matteofigus — APPROVED** [review]
LGTM (left a nitpick comment about gitignore). A second review by someone more familiar with the codebase would be appreciated anyways, for extra safety. Your call 👍

File: .idea/aws.xml (hunk 5 above)
Comment (svozza) [reply .idea/aws.xml:8]: This is very weird, I have that whole folder where that file lives excluded in my `.gitignore` folder.

**Pull request merged**